Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/Saml2AllowNoSubjectAssertionSpecValidator.java

```diff
@@ -43,32 +49,65 @@ public Saml2AllowNoSubjectAssertionSpecValidator() {
         super();
     }
 
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * gov.hhs.fha.nhinc.callback.cxf.Saml2ExchangeAuthFrameworkValidator#validateAssertion(org.apache.wss4j.common.saml
+     * .SamlAssertionWrapper)
+     */
+    @Override
+    protected void validateAssertion(final SamlAssertionWrapper samlAssertion) throws WSSecurityException {
+        try {
+            validateSubject(samlAssertion.getSaml2());
+        } catch (final WSSecurityException e) {
+            LOG.error("Unable to validate Assertion ", e);
```

Comment:
@mhpnguyen As a rule of thumb, you want to log a stack trace ONLY where you plan to handle it.  Anywhere else in the call stack, it's usually appropriate to either:
- Not log the error at all ("Unable to validate Assertion" is implicitly understood by the stack trace coming from the function "validateAssertion")
- Log the error message, but NOT the stack ("Unable to.... {}", e.getLocalizedMessage())

In the rare instances where having the stack trace logged along the way would be truly invaluable to troubleshooting efforts, consider throwing it in at the trace level (to prevent the logs from becoming too noisy at the debug level).